login/service.go

```diff
@@ -840,6 +843,30 @@ func (keycloak *KeycloakOAuthProvider) equalsKeycloakUserProfileAttributes(ctx c
 	return profileEqual, nil
 }
 
+func fillUserFromResponse(userProfile oauth.UserProfile, identity *account.Identity) (bool, error) {
+	isChanged := false // will go away in future.
+	identity.ID, _ = uuid.FromString(userProfile.Subject)
+	identity.User.FullName = name.GenerateFullName(&userProfile.GivenName, &userProfile.FamilyName)
+	identity.User.Email = userProfile.Email
+
+	identity.User.Company = userProfile.Company
+	identity.User.EmailVerified = userProfile.EmailVerified
+	identity.Username = userProfile.Username
+	if identity.User.ImageURL == "" {
+		image, err := generateGravatarURL(userProfile.Email)
+		if err != nil {
+			log.Warn(nil, map[string]interface{}{
```

Comment:
why `log.Warn` and not `log.Error`, or even better, what about logging the errors at the top/end caller, rather than when emitting it ? otherwise, I think we'll have more or less the same error multiple times in the logs.

Reply:
Since we lookup the user image from gravatar, there could be an external service disruption which could cause an error. In that case, we are not gonna stall the login/signup procedure, hence we would would just log and move on.